### DISCUSSION TITLE
Opaque texture - not mixed with material color?

### DISCUSSION THREAD

**Discussion by diggitumb** [assimp/assimp#4080] [Q&A] [answered]
In the documentation the combination of the first texture of the diffuse texture stack with the material color is discribed as:
`FirstTexLayerResultColor = MaterialColor <texOp> (strength * TextureColor)`
texOp is one of aiTextureOp (MULT,ADD,SUB,DIV,SMOOTHEADD,SIGNEDADD).

Imagine a grey concrete wall overpainted with a graffiti. The graffiti colors should not be mixed with the grey wall color.
Which combination of material/texture properties leads to a replacement of the underlying material color with the texture color?

Currently my fragment shader does something like
`FirstTexLayerResultColor = ((1-strength) * MaterialColor) <texOp> (strength * TextureColor)`
With strength=1.0 this results in an opaque texture.

**Comment by kimkulling** [comment, marked as answer]
You need to do this in your fragment-shader. There is no option in assimp to perform the blend-operation.
